src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs

```diff
@@ -368,45 +369,24 @@ public T WithCorrelationId(Guid correlationId)
         ///  PoP tokens are similar to Bearer tokens, but are bound to the HTTP request and to a cryptographic key, which MSAL can manage on Windows.
         ///  See https://aka.ms/msal-net-pop
         /// </summary>
-        /// <param name="httpRequestMessage">An HTTP request to the protected resource which requires a PoP token. The PoP token will be cryptographically bound to the request.</param>
+        /// <param name="popAuthenticationConfiguration"> </param>
         /// <remarks>
         /// <list type="bullet">
-        /// <item>This is an experimental API. The method signature may change in the future without involving a major version upgrade.</item>
         /// <item> An Authentication header is automatically added to the request</item>
         /// <item> The PoP token is bound to the HTTP request, more specifically to the HTTP method (GET, POST, etc.) and to the Uri (path and query, but not query parameters). </item>
         /// <item>MSAL creates, reads and stores a key in memory that will be cycled every 8 hours.</item>
         /// </list>
         /// </remarks>
-        public T WithProofOfPosession(HttpRequestMessage httpRequestMessage)
+        public T WithProofOfPosession(PopAuthenticationConfiguration popAuthenticationConfiguration)
         {
-            var defaultCryptoProvider = ServiceBundle.PlatformProxy.GetDefaultPoPCryptoProvider();
-            return WithProofOfPosession(httpRequestMessage, defaultCryptoProvider);
-        }
-
-        // Allows testing the PoP flow with any crypto. Consider making this public.
-        internal T WithProofOfPosession(HttpRequestMessage httpRequestMessage, IPoPCryptoProvider popCryptoProvider)
-        {
-            if (!ServiceBundle.Config.ExperimentalFeaturesEnabled)
-            {
-                throw new MsalClientException(
-                    MsalError.ExperimentalFeature,
-                    MsalErrorMessage.ExperimentalFeature(nameof(WithProofOfPosession)));
-            }
-
-            if (httpRequestMessage is null)
-            {
-                throw new ArgumentNullException(nameof(httpRequestMessage));
-            }
-
-            if (popCryptoProvider == null)
+            if (popAuthenticationConfiguration is null)
             {
-                throw new ArgumentNullException(nameof(popCryptoProvider));
+                throw new ArgumentNullException(nameof(popAuthenticationConfiguration));
             }
 
-            CommonParameters.AddApiTelemetryFeature(ApiTelemetryFeature.WithPoPScheme);
-            CommonParameters.AuthenticationScheme = new PoPAuthenticationScheme(httpRequestMessage, popCryptoProvider);
-
-            return this as T;
+            CommonParameters.UsingProofOfPossesion = true;
```

Comment:
Is this really needed? If we have a POP config, we must be doing POP.

Reply:
In fact, why don't you just create a new `PoPAuthenticationScheme` here directly?